compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/window/ComposeWindow.web.kt

```diff
@@ -681,10 +708,52 @@ fun ComposeViewport(
         left = "0"
     }
 
+    val a11yContainerElement = document.createElement("div") as HTMLDivElement
+
+    if (a11YConfiguration.isA11YEnabled()) {
+        layerRoot.appendChild(a11yContainerElement)
+        a11yContainerElement.style.apply {
+            position = "absolute"
+            top = "0"
+            left = "0"
+        }
+    }
+
     ComposeWindow(
         canvas = canvas,
         interopContainerElement = interopContainerElement,
+        a11yContainerElement = a11yContainerElement,
         content = content,
+        a11YConfiguration = a11YConfiguration,
         state = DefaultWindowState(viewportContainer)
     )
+}
+
+/**
+ * Represents the configuration for accessibility (a11y) features in a Compose-based application.
+ */
+@ExperimentalComposeUiApi
+sealed class A11YConfiguration {
+    internal object Disabled : A11YConfiguration()
+    internal object Enabled : A11YConfiguration()
+
+    internal fun isA11YEnabled(): Boolean {
+        return this == Enabled
+    }
+}
```

Comment:
why do we design it as a sealed class instead of a simple enum?

Reply:
Changed it again according to our discussion:

```
class ComposeViewportConfiguration internal constructor() {

    @ExperimentalComposeUiApi
    var isA11YEnabled: Boolean = true
}
```